### ISSUE TITLE
PermissionError when writing executable in --onefile mode

### ISSUE THREAD

**Issue by DervishD** [Nuitka/Nuitka#1445]
* Nuitka version, full Python version, flavor, OS, etc. as output by this
  command (it does more than you think, and we are adding more all the time):
```
0.7.1
Commercial: None
Python: 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
Flavor: Unknown
Executable: C:\Users\rauln\Desktop\vsm\venv\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```

* How did you install Nuitka and Python

`pip install nuitka` on `virtualenv`
 
* The specific PyPI names and versions
```
Nuitka==0.7.1
zstandard==0.17.0
```

* Many times when you get an error from Nuitka, your setup may be special

 Even a `print("hello world")` program does not work.

* Provide in your issue the Nuitka options used

`python -m nuitka --onefile test.py`

When generating the `.exe` file on Windows, probably due to a race condition provoked by the antivirus, but it happens with the antivirus disabled, the freezing generates the following error:
```
Nuitka-Options:INFO: Used command line options: --onefile test.py
Nuitka:INFO: Starting Python compilation with Nuitka '0.7.1' on Python '3.9' commercial None.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'socket' added '_socket'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'pathlib' added 'ntpath,posixpath'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'importlib.metadata' added 'email'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'ctypes' added '_ctypes'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'locale' added '_bootlocale'.
Nuitka-Plugins:INFO: implicit-imports: Implicit dependencies of module 'encodings' added 'encodings.idna,encodings.base64_codec,encodings.hex_codec,encodings.rot_13'.
Nuitka:INFO: Completed Python level compilation and optimization.
Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: gcc (gcc).
Nuitka-Scons:INFO: Backend linking program (no progress information available).
Nuitka-Scons:INFO: Compiled 10 C files using ccache.
Nuitka-Scons:INFO: Cached C files (using ccache) with result 'cache hit': 10
Nuitka-Postprocessing:INFO: Creating single file from dist folder, this may take a while.
Nuitka-Onefile:INFO: Running bootstrap binary compilation via Scons.
Nuitka-Scons:INFO: Onefile C compiler: gcc (gcc).
Nuitka-Scons:INFO: Onefile linking program (no progress information available).
Nuitka-Scons:INFO: Compiled 1 C files using ccache.
Nuitka-Scons:INFO: Cached C files (using ccache) with result 'cache hit': 1
Nuitka-Onefile:INFO: Keeping onefile build directory 'test.onefile-build'.
Nuitka-Onefile:INFO: Using compression for onefile payload.
Traceback (most recent call last):
  File "C:\Users\rauln\Desktop\vsm\venv\lib\site-packages\nuitka\__main__.py", line 137, in <module>
    main()
  File "C:\Users\rauln\Desktop\vsm\venv\lib\site-packages\nuitka\__main__.py", line 123, in main
    MainControl.main()
  File "C:\Users\rauln\Desktop\vsm\venv\lib\site-packages\nuitka\MainControl.py", line 974, in main
    packDistFolderToOnefile(dist_dir, binary_filename)
  File "C:\Users\rauln\Desktop\vsm\venv\lib\site-packages\nuitka\freezer\Onefile.py", line 72, in packDistFolderToOnefile
    packDistFolderToOnefileBootstrap(onefile_output_filename, dist_dir)
  File "C:\Users\rauln\Desktop\vsm\venv\lib\site-packages\nuitka\freezer\Onefile.py", line 408, in packDistFolderToOnefileBootstrap
    runOnefileCompressor(
  File "C:\Users\rauln\Desktop\vsm\venv\lib\site-packages\nuitka\freezer\Onefile.py", line 351, in runOnefileCompressor
    attachOnefilePayload(
  File "C:\Users\rauln\Desktop\vsm\venv\lib\site-packages\nuitka\tools\onefile_compressor\OnefileCompressor.py", line 73, in attachOnefilePayload
    with open(onefile_output_filename, "ab") as output_file:
PermissionError: [Errno 13] Permission denied: 'test.exe'
```

It is probably this error [same error in `pyinstaller`](https://github.com/pyinstaller/pyinstaller/issues/6467) (yes, I reported the same there, as I mentioned before I'm testing many similar tools).

Changing the suffix from `.exe` to something like `.none` makes the process work, just like in `pyinstaller` tests.

Again, this is not a bug in Nuitka, but an interaction between how Nuitka creates the executable and the weird behaviour of Windows default antivirus, which is locking the executable, no doubt. Still, maybe Nuitka should handle this situation.

And again, thanks 😃

**Comment by kayhayen** [comment]
I think for updating the resources, we got already extra balls, probably needed here as well. Having a non-exe suffix is too annoying for me to handle. I am 100% positive it's possible to disable AV on Windows, it's just not a wide-spread skill to do it properly, and as such becomes our business too.

**Comment by DervishD** [comment]
Disabling the default AV is not that hard, the problem is that the locking seems to persist even with the AV disabled. I've even tried rebooting, but it does not always work, I'm afraid.

And no, changing the suffix is not the solution, just a partial evidence that maybe it's the AV the one to blame.

If you want me to do any tests, just tell me and I'll be very happy to help. As I said, the least I can do.

**Comment by kayhayen** [comment]
I am preparing something hotfix worthy for later today.

**Comment by DervishD** [comment]
I'll set up a virtualenv with nuitka in my home computer, then, to be able to test everything ASAP. Thanks A LOT, Kay.

**Comment by kayhayen** [comment]
There you go, https://nuitka.net/doc/factory.html

This is going to be in 0.7.2 hotfix release, coming relatively.

**Comment by kayhayen** [comment]
It will give warnings, wait a second, then retry. For the resources' attachment, that has meant I never heard of it again. For the onefile, resources are attached before, so maybe that's because you don't have an icon or something, or because after changing the resources, AV wakes up to it again, doesn't matter though.

It has to be said, that in all my CI, never have I had the message appear, but there were always some users, some of which had it occur reproducible and also swear to have deactivated AV. Maybe it's a sort of race condition, that only some people loose.

Anyway, Nuitka can recommend to disable AV, but honestly that's not good to ask, and I have a desire that people of all skill levels can use it out of the box. So this gets the failure rate even closer to zero for them, I am happy.

**Comment by DervishD** [comment]
IT WORKS! Looks like that one second delay makes all the difference.

I've tried adding the building folders, the virtualenv folder, the executable file, etc. to Windows Defender exception list, I've tried disabling the AV, only parts of it (e.g. the real-time protection) to no avail, and this is a problem that is pestering other Python software I use, because the AV fires some kind of race condition.

BTW, I asked a friend, who has a slightly faster computer than mine, and he cannot reproduce the problem, and I can't reproduce it neither on my wife's computer. So right know I can only reproduce it in two computers. It's quite weird.

I tested with the simple 'hello world' example and with a more complicated script including third party libraries, and it works for both of them. I have to perform further tests to check if the executable is properly self-confined, but so far everything works.

And I agree, recommending disabling AV for building is not a good idea, in fact I would not even mention it in the warning, because if this happens for other reason some people may still disable the AV for good, I don't know.
